src/document/range.js

```diff
@@ -72,6 +72,192 @@ CKEDITOR.define( [ 'document/positioniterator', 'document/position' ], ( Positio
 		}
 
 		/**
+		 * Creates and returns a new instance of {@link document.Range}
+		 * that is equal to this {@link document.Range range}.
+		 *
+		 * @returns {document.Position} Cloned {@link document.Range range}.
+		 */
+		clone() {
+			return new Range( this.start.clone(), this.end.clone() );
+		}
+
+		/**
+		 * Checks whether this {document.Range range} contains given (document.Position position}.
+		 *
+		 * @param {document.Position} position Position to check.
+		 * @returns {boolean} True if given {document.Position position} is contained.
+		 */
+		containsPosition( position ) {
+			return position.isAfter( this.start ) && position.isBefore( this.end );
+		}
+
+		/**
+		 * Returns an array containing one or two {document.Range ranges} that are results of transforming this
+		 * {document.Range range} by inserting `howMany` nodes at `insertPosition`. Two {document.Range ranges} are
+		 * returned if the insertion was inside this {document.Range range}.
+		 *
+		 * Examples:
+		 * 	let range = new Range( new Position( [ 2, 7 ], root ), new Position( [ 4, 0, 1 ], root ) );
```

Comment:
Indentation: spaces -> 2tabs. Or we can fix it later after we'll figure out which doc generator we want to use. But in general, since we use MD, 2tabs are the minimum to make it a code block in this flavour. We can later agree though to use the GH's alternative (triple backticks). So... ignore this is you're lazy :D.